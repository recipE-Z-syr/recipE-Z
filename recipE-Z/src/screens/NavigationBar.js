import React, { Component } from 'react';
import { Button, StyleSheet, Text, TextInput, View, SafeAreaView, ScrollView, Keyboard, TouchableOpacity, Image } from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

class NavigationBar extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {navigation} = this.props;
    return (
      <View style={{flex: 1, flexDirection: 'row-reverse', position: 'absolute', top: 30, right: 30}}>
      <TouchableOpacity onPress={() => navigation.navigate('Profile')}>
        <Image
          style={{width: 25, height: 25, marginLeft: 20}}
          source={require('../img/profile-color.png')}
        />
      </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('RecipeBook')}>
          <Image
            style={{width: 25, height: 25, marginLeft: 20}}
            source={require('../img/favorites.png')}
          />
        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Upload')}>
          <Image
            style={{width: 25, height: 25, marginLeft: 20}}
            source={require('../img/create.png')}
          />
        </TouchableOpacity>


      </View>
   ); //end return
 }//end render
}//end class

export default NavigationBar;

const styles = StyleSheet.create({
});
