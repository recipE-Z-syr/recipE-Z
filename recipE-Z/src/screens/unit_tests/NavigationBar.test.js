//NavigationBar.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from "enzyme";

import NavigationBar from '../NavigationBar';

describe('<NavigationBar />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NavigationBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
});

test('Navigates to Profile', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<NavigationBar {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(0)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Profile')
})

test('Navigates to Go to RecipeBook', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<NavigationBar {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(1)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('RecipeBook')
})

test('Navigates to Go to Upload', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<NavigationBar {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(2)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Upload')
})