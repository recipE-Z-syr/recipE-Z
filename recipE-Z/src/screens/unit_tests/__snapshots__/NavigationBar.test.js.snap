// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row-reverse",
      "position": "absolute",
      "right": 30,
      "top": 30,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 25,
          "marginLeft": 20,
          "width": 25,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 25,
          "marginLeft": 20,
          "width": 25,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 25,
          "marginLeft": 20,
          "width": 25,
        }
      }
    />
  </View>
</View>
`;
